### ISSUE TITLE
Ansible 2.3.0.0 uri response would replace character b with ********

### ISSUE THREAD

**Issue by yunj** [ansible/ansible#23671]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

uri

##### ANSIBLE VERSION

```
2.3.0.0
```

##### CONFIGURATION

[defaults]
hostfile = hosts
roles_path = .galaxy:roles

##### OS / ENVIRONMENT

Windows 8.1

##### SUMMARY

Ansible 2.2.0.0 uri module behaves as expected but after upgrading to 2.3.0.0 uri response would replace character b with ********

##### STEPS TO REPRODUCE

Access some Rest API which would return application/xml containing, e.g., 1.0.0.beta. Please see the attachment as below.
![uri response](https://cloud.githubusercontent.com/assets/18545475/25104112/35a32316-2413-11e7-81e0-4733d35925c0.PNG)

```yaml
- name: Retrieve uploaded solutions
  uri:
    url: http://ip:port/path/to/resource
    user: admin
    password: b
    force_basic_auth: yes
    return_content: yes
  register: result
  until: result.status == 200
```

##### EXPECTED RESULTS

1.0.0.beta
321 bytes
##### ACTUAL RESULTS

1.0.0.********eta
321 ********ytes

```

```

**Comment by abadger** [comment]
tldr; Do not set your password to "b".

Ansible will hide any data that is passed in via a module parameter marked no_log if it occurs anywhere in the return results.  This is because Ansible itself does not know how the results got there, it only knows that the value of the results contains information that has been marked as being secret.  If I create a new user account on a web service and the web service sends the password I passed in as part of its reply, Ansible doesn't know that the web service is sending back the password.  It only knows that the value matches with the value passed in in the password parameter to the module.

It's better for ansible to err on the side of caution than to accidentally reveal information that is supposed to be secret so I'm afraid we aren't going to change this.  If this bothers you, change your password to something secure and not based on a dictionary word.

Thanks, Toshio

**Comment by yunj** [comment]
Thanks for your information. My script used to work for Ansible 2.2.0.0 then fails after upgrading to 2.3.0.0. I would change the password to something else.
